lib/json_schema/schema_error.rb

```diff
@@ -18,12 +18,13 @@ def to_s
   end
 
   class ValidationError < SchemaError
-    attr_accessor :path, :sub_errors
+    attr_accessor :path, :sub_errors, :data
 
-    def initialize(schema, path, message, type, sub_errors = nil)
+    def initialize(schema, path, message, type, sub_errors = nil, data: nil)
```

Comment:
+1 to an optional argument here. We can probably modify this interface to make `sub_errors` also optional at some point for general housekeeping. No need to change it for now though!